src/test/java/teammates/test/cases/datatransfer/FeedbackResponseDetailsTest.java

```diff
@@ -281,4 +290,97 @@ public void testCreateResponseDetails() {
+     */
+
+    @Test
+    public void testStudentQuestionResultsStatisticsGeneratingMethods() {
+        // Load databundle with NUMSCALE questions- and session-info
+        DataBundle db = loadDataBundle("/FeedbackSessionQuestionTypeTest.json");
+
+        /**
+         * Object setup for method-calls.
+         * Such as a course roster with students and instructors that partake in certain courses
+         * and some data regarding the student responses to a question.
+         */
+        FeedbackResponseAttributes fra = db.feedbackResponses.get("response1ForQ1S3C1");
+        assertNotNull(fra);
+        FeedbackQuestionAttributes fqa = db.feedbackQuestions.get("qn1InSession3InCourse1");
+        assertNotNull(fqa);
+
+        FeedbackSessionAttributes fsa = db.feedbackSessions.get("numscaleSession");
+        List<StudentAttributes> students = new ArrayList<StudentAttributes>();
+        students.add(db.students.get("student1InCourse1"));
+        students.add(db.students.get("student2InCourse1"));
+        students.add(db.students.get("student3InCourse1"));
+        students.add(db.students.get("student4InCourse1"));
+        students.add(db.students.get("student5InCourse1"));
+        students.add(db.students.get("student1InCourse2"));
+        students.add(db.students.get("student2InCourse2"));
+        students.add(db.students.get("student1InCourseWithSections"));
+        List<InstructorAttributes> instructors = new ArrayList<InstructorAttributes>();
+        instructors.add(db.instructors.get("instructor1OfCourse1"));
+        instructors.add(db.instructors.get("instructor2OfCourse1"));
+        instructors.add(db.instructors.get("instructor1OfCourse2"));
+        instructors.add(db.instructors.get("instructor2OfCourse2"));
+        instructors.add(db.instructors.get("instructor1OfCourseWithSections"));
+
+        HashMap<String, FeedbackQuestionAttributes> mapFqa = new HashMap<>();
+        mapFqa.put(fqa.getId(), fqa);
+
+        CourseRoster cr = new CourseRoster(students, instructors);
+        FeedbackSessionResultsBundle bundle = new FeedbackSessionResultsBundle(fsa, mapFqa, cr);
+
+        for (StudentAttributes student : students) {
+            bundle.emailTeamNameTable.put(student.email, student.team);
+            bundle.emailNameTable.put(student.email, student.name);
+            bundle.emailLastNameTable.put(student.email, student.lastName);
+        }
+
+        for (InstructorAttributes instructor : instructors) {
+            bundle.instructorEmailNameTable.put(instructor.email, instructor.name);
+        }
+
+        ArrayList<FeedbackResponseAttributes> fraList = new ArrayList<>();
+        fraList.add(fra);
+        /*
+         * Test if the objects created during setup are null
+         */
+        assertNotNull(fraList);
+        assertNotNull(fsa);
+        assertNotNull(mapFqa);
+        assertNotNull(cr);
+        assertNotNull(bundle);
```

Comment:
Are there actual usages of these non-null assertions? If any of them are null, you're going to have NPEs from long before any of these is reached.